l3experimental/l3opacity/l3opacity.dtx

```diff
@@ -117,14 +125,13 @@
 \cs_new_protected:Npn \opacity_stroke:n #1
   { \@@_select:nN {#1} \@@_backend_stroke:n }
 \cs_new_protected:Npn \@@_select:nN #1#2
-  { \exp_args:Ne \@@_select_aux:nN { \fp_eval:n {#1} } #2 }
-\cs_new_protected:Npn \@@_select_aux:nN #1#2
   {
+    \fp_set:Nn \l_@@_tmpa_fp { #1 }
```

Comment:
Why set a var rather than `e`-type expand `\fp_eval:n` using an auxiliary?

Reply:
Because `\fp_eval:n { <fp expr> }` is slower than `\fp_eval:n { <fp var> }`, provided `<fp var>` is set by `\fp_set:Nn <fp var> { <fp expr> }`.

Since here `#1` will be evaluated two or three times, store the parsed form in an fp-var saves some time.